pkgs/by-name/do/docmost/bcrypt-lib.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+}:
+let
+  # Adapted from pkgs/server/peertube/default.nix
+  hashes = {
+    x86_64-linux =
+      if stdenv.hostPlatform.isMusl then
+        "sha256-m2nJM8f6Vz9n3yT/QSmLBCbjDXMWmwBgIJ4lLyO6g9s="
+      else
+        "sha256-bDXV6vSlFbTHs3MAil4VMcKq7Xq5V6LHs9RxKFgqxus=";
+    aarch64-linux =
+      if stdenv.hostPlatform.isMusl then
+        "sha256-5CiYal1s4oPqWwC1POu95BLQPW+N950sg9po2T4txCw="
+      else
+        "sha256-DvO9ChcMZTJ/2OrtdqVoq7sDhuLOjO1iq/sGRPunw4o=";
+    x86_64-darwin = "sha256-5DIivuMnjmKHXgGA6LH1IblMLe+EcbyXRMewjgzH3x0=";
+    aarch64-darwin = "sha256-tsonr8qfJnpfcjPoutmnx3pOT1NoE1hFM0CqSVDQW6c=";
+    x86_64-windows = "sha256-Pu2C2WWlNae00vQ19reX8LZ9Nr/oajtZLzinTy7geIo=";
+  };
+
+  os =
+    {
+      linux = "linux";
+      darwin = "darwin";
+      windows = "win32";
+    }
+    .${stdenv.hostPlatform.parsed.kernel.name};
+
+  arch =
+    {
+      x86_64 = "x64";
+      aarch64 = "arm64";
+    }
+    .${stdenv.hostPlatform.parsed.cpu.name};
+
+  libc = if stdenv.hostPlatform.isLinux then stdenv.hostPlatform.libc else "unknown";
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "bcrypt-lib";
+  version = "5.1.1";
+
+  src = fetchzip {
+    url = "https://github.com/kelektiv/node.bcrypt.js/releases/download/v${finalAttrs.version}/bcrypt_lib-v${finalAttrs.version}-napi-v3-${os}-${arch}-${libc}.tar.gz";
+
+    hash = hashes.${stdenv.hostPlatform.system};
+  };
+
+  dontBuild = true;
```

Comment:
Please add a comment explaining why we don't build from source

Reply:
Good question. I think I only did it this way because I saw another package that downloads bcrypt and didn't question it further — also, I don't know where you would put this since `nodePackages` isn't exactly designed to be extensible

Reply:
🤔 Is this an upstream bug or is it just not tested against Clang? Both could happen, TBH. Maybe try GCC?

Reply:
The log is from ofborg. If you want to try something, just push to this PR.

Reply:
I don't own any Macs 🤷🏼‍♀️